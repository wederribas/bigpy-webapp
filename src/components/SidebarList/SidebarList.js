import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import Divider from '@material-ui/core/Divider'
import SearchIcon from '@material-ui/icons/Search'
import WhatshotIcon from '@material-ui/icons/Whatshot'
import HighlightIcon from '@material-ui/icons/Highlight'
import InfoIcon from '@material-ui/icons/Info'

const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper
  }
})

function SidebarList(props) {
  const { classes } = props

  return (
    <div id={'Testing'} className={classes.root}>
      <List component="nav">
        <ListItem button>
          <ListItemIcon>
            <SearchIcon />
          </ListItemIcon>
          <ListItemText primary="Search" />
        </ListItem>
        <ListItem button>
          <ListItemIcon>
            <WhatshotIcon />
          </ListItemIcon>
          <ListItemText primary="Top 10" />
        </ListItem>
        <ListItem button>
          <ListItemIcon>
            <HighlightIcon />
          </ListItemIcon>
          <ListItemText primary="Insights" />
        </ListItem>
      </List>
      <Divider />
      <List component="nav">
        <ListItem button>
          <ListItemIcon>
            <InfoIcon />
          </ListItemIcon>
          <ListItemText primary="About" />
        </ListItem>
      </List>
    </div>
  )
}

SidebarList.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(SidebarList)
