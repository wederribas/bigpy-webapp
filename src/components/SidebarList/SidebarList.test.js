import React from 'react'
import { render } from 'react-testing-library'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import SidebarList from './SidebarList'

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
})

test('Sidebar list', () => {
  const sidebar = render(
    <MuiThemeProvider theme={theme}>
      <SidebarList />
    </MuiThemeProvider>
  )
  const { getByText } = sidebar

  expect(getByText('Search')).toHaveTextContent('Search')
  expect(getByText('Top 10')).toHaveTextContent('Top 10')
  expect(getByText('Insights')).toHaveTextContent('Insights')
  expect(getByText('About')).toHaveTextContent('About')
})
