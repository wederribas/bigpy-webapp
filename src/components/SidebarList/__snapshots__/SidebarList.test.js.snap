// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar list 1`] = `
<div
  class="SidebarList-root-1"
>
  <nav
    class="MuiList-root-3 MuiList-padding-4"
  >
    <a
      class="SidebarList-noDecoration-2"
      href="/"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15 MuiListItem-selected-17"
        data-testid="list-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root-21"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-22"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root-31"
        >
          <span
            class="MuiTypography-root-37 MuiTypography-subtitle1-55 MuiListItemText-primary-34"
          >
            Search
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-76"
        />
      </div>
    </a>
    <a
      class="SidebarList-noDecoration-2"
      href="/top10"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15"
        data-testid="list-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root-21"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-22"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root-31"
        >
          <span
            class="MuiTypography-root-37 MuiTypography-subtitle1-55 MuiListItemText-primary-34"
          >
            Top 10
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-76"
        />
      </div>
    </a>
    <a
      class="SidebarList-noDecoration-2"
      href="/insights"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15"
        data-testid="list-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root-21"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-22"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <defs>
              <path
                d="M0 0h24v24H0V0z"
                id="a"
              />
            </defs>
            <path
              d="M6 14l3 3v5h6v-5l3-3V9H6zm5-12h2v3h-2zM3.5 5.875L4.914 4.46l2.12 2.122L5.62 7.997zm13.46.71l2.123-2.12 1.414 1.414L18.375 8z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root-31"
        >
          <span
            class="MuiTypography-root-37 MuiTypography-subtitle1-55 MuiListItemText-primary-34"
          >
            Insights
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-76"
        />
      </div>
    </a>
  </nav>
  <hr
    class="MuiDivider-root-72"
  />
  <nav
    class="MuiList-root-3 MuiList-padding-4"
  >
    <a
      class="SidebarList-noDecoration-2"
      href="/about"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15"
        data-testid="list-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root-21"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-22"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root-31"
        >
          <span
            class="MuiTypography-root-37 MuiTypography-subtitle1-55 MuiListItemText-primary-34"
          >
            About
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-76"
        />
      </div>
    </a>
  </nav>
</div>
`;
